runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -542,13 +543,16 @@ setCallSiteTargetImpl(J9VMThread *currentThread, jobject callsite, jobject targe
 		MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
 
 		/* Check for MutableCallSite modification. */
-		J9JITConfig* jitConfig = currentThread->javaVM->jitConfig;
-		J9UTF8 *clazzNameUTF8 = J9ROMCLASS_CLASSNAME(clazz->romClass);
-		U_8 *clazzName = J9UTF8_DATA(clazzNameUTF8);
-		UDATA clazzLength = J9UTF8_LENGTH(clazzNameUTF8);
+		J9JITConfig* jitConfig = javaVM->jitConfig;
+		J9Class *mcsClass = vmFuncs->hashClassTableAt(
+			javaVM->systemClassLoader,
+			(U_8*)"java/lang/invoke/MutableCallSite",
+			LITERAL_STRLEN("java/lang/invoke/MutableCallSite"));
+
 		if (!isVolatile /* MutableCallSite uses setTargetNormal(). */
 		&& NULL != jitConfig
-		&& J9UTF8_LITERAL_EQUALS(clazzName, clazzLength, "java/lang/invoke/MutableCallSite")
+		&& NULL != mcsClass
+		&& VM_VMHelpers::inlineCheckCast(clazz, mcsClass)
```

Comment:
A bit of nitpick but vm convention is each clause of the if should be individually bracketed.  ie:
```
&& (NULL != xxxx)
&& VM_VMHelpers::inlineCheckCast(clazz, mcsClass)
```

Reply:
OK, I'll parenthesize the null tests